lib/util.c

```diff
@@ -155,3 +155,21 @@ unsigned int RoundTo(const unsigned int num, const unsigned int what)
     return what * ((num + what -1)/what);
 }
 
+/**
+ * to zero a buffer in a more secure manner meant for a handful of purposes.
+ * e.g. for password matters.
+ * The compiler can optimize away a memset call to gain performance here
+ * making sure it does not occur.
+ *
+ * memset_s is designed for such cases otherwise putting the
+ * address in a volatile pointer avoid gcc's likes doing optimizations.
+ */
+void ClearMem(void *dst, size_t len)
```

Comment:
This is not how we do portability in Squid. We use the relevant system call unconditionally in the main code and provide an implementation in libcompat for OS where it is not available.  See the compat/memrchr files for an example.